### PR TITLE
add predicate for use fhir service

### PR DESCRIPTION
**Pull request by peizhou298**
## Description
This PR is add a predicate arguments in UseFhirServer function to enable the service filter the health check services it want to call, without it every registered services will be called in each health check.

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- [ ] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [ ] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [ ] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [ ] CI is green before merge
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Api/Registration/FhirServerApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -23,10 +24,21 @@ public static class FhirServerApplicationBuilderExtensions
         /// <param name="app">The application builder instance.</param>
         /// <returns>THe application builder instance.</returns>
         public static IApplicationBuilder UseFhirServer(this IApplicationBuilder app)
+        {
+            return app.UseFhirServer(null);
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Api/Registration/FhirServerApplicationBuilderExtensions.cs:28]: Nit: Should this method have an EnsureArg for app like the other UseFhirServer method?

**Review by LTA-Thinking — APPROVED** [review]

**Comment by LTA-Thinking** [comment]
This looks good. Where is this planned to be used?

File: src/Microsoft.Health.Fhir.Api/Registration/FhirServerApplicationBuilderExtensions.cs [hunk 2]
```diff
@@ -23,10 +24,21 @@ public static class FhirServerApplicationBuilderExtensions
...
+        }
+
+        /// <summary>
+        /// Adds FHIR server functionality to the pipeline with health check filter.
+        /// </summary>
+        /// <param name="app">The application builder instance.</param>
+        /// <param name="predicate">The predicate used to filter health check services.</param>
+        /// <returns>THe application builder instance.</returns>
+        public static IApplicationBuilder UseFhirServer(this IApplicationBuilder app, Func<HealthCheckRegistration, bool> predicate)
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Api/Registration/FhirServerApplicationBuilderExtensions.cs:37]: Could these two methods just be collapsed with a default value?
```suggestion
        public static IApplicationBuilder UseFhirServer(this IApplicationBuilder app, Func<HealthCheckRegistration, bool> predicate = null)
```

**Comment by peizhou298** [comment]
> 



> This looks good. Where is this planned to be used?

This is used in Paas, the scenario is we need a to add a new health check service there, and we don't want default health check endpoint(/health/check) contains this service and added endpoint(health/check) only contain this, so we need a predicate to achieve it.

File: src/Microsoft.Health.Fhir.Api/Registration/FhirServerApplicationBuilderExtensions.cs (hunk 2 above)
Comment (peizhou298) [reply src/Microsoft.Health.Fhir.Api/Registration/FhirServerApplicationBuilderExtensions.cs]: yes, it could. Already changed to that format.

**Comment by brendankowitz** [comment]
CI was not green before merge

**Pull request merged**